File: src/main/java/ash/java/graphql/types/TmdbObjectType.java
package ash.java.graphql.types;

import graphql.annotations.GraphQLAnnotations;
import graphql.annotations.GraphQLField;
import graphql.schema.GraphQLObjectType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public abstract class TmdbObjectType {

    protected Logger log = LoggerFactory.getLogger(this.getClass());

    @GraphQLField
    protected Integer id;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public GraphQLObjectType getGraphQlType() {
        GraphQLObjectType objectType = null;
        try {
            objectType = GraphQLAnnotations.object(this.getClass());
        } catch (IllegalAccessException | InstantiationException | NoSuchMethodException e) {
            log.error("Exception while trying to produce GraphQL type for class", e);
        }

        return objectType;
    }
}

File: src/main/kotlin/ash/kotlin/graphql/types/TmdbObjectType.kt
package ash.kotlin.graphql.types

import graphql.annotations.GraphQLAnnotations
import graphql.annotations.GraphQLField
import graphql.schema.GraphQLFieldDefinition
import graphql.schema.GraphQLInterfaceType
import graphql.schema.GraphQLObjectType
import org.slf4j.Logger
import org.slf4j.LoggerFactory

abstract class TmdbObjectType(@GraphQLField val id: Int) {
    protected val log: Logger = LoggerFactory.getLogger(this.javaClass)

    fun getGraphQlType(): GraphQLObjectType {
        return try {
            GraphQLAnnotations.`object`(this.javaClass)
        } catch (e: ReflectiveOperationException) {
            log.error("Exception while trying to produce GraphQL type for class", e)
            GraphQLObjectType("", "", listOf<GraphQLFieldDefinition>(), listOf<GraphQLInterfaceType>())
        }
    }
}